packages/material-ui/src/Accordion/Accordion.js

```diff
@@ -98,6 +136,15 @@ const Accordion = React.forwardRef(function Accordion(props, ref) {
     ...other
   } = props;
 
+  const styleProps = {
+    ...props,
+    square,
+    disabled,
+    expanded: expandedProp,
```

Comment:
This is not the correct value for `expanded`. See on line 148 we are using the value coming from `useControlled`. Please move the definition of the `styleProps` after that and use the `expanded` from the hook result.